setup.py

```diff
@@ -197,13 +197,15 @@ def run(self):
     'testpath', 
 ]
 
-extras_require = setuptools_args['extras_require'] = {
+extra_requirements = {
     # FIXME: tests still require nose for some utility calls,
     # but we are running with pytest
-    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client'],
+    'test': ['pytest', 'pytest-cov', 'nose', 'ipykernel', 'jupyter_client>=4.2'],
     'serve': ['tornado>=4.0'],
     'execute': ['jupyter_client>=4.2'],
 }
+extra_requirements['all'] = sum(extra_requirements.values(), [])
+extras_require = setuptools_args['extras_require'] = extra_requirements
```

Comment:
I don't think we need to have this called both `extra_requirements` and `extras_require`. There's nothing in the code after this using it by either name; its only in `setuptools_args` that it's used.